datastore/transaction.go

```diff
@@ -196,6 +196,17 @@ func (c *Client) RunInTransaction(ctx context.Context, f func(tx *Transaction) e
 	settings := newTransactionSettings(opts)
 	for n := 0; n < settings.attempts; n++ {
 		tx, err := c.newTransaction(ctx, settings)
+
+		// Ensure that any panic has a corresponding rollback
+		defer func(t *Transaction) {
```

Comment:
This seems likely like fixing a symptom and not the problem. When does this code panic? We should make it so it does not and return an error if needed.

Reply:
This library has a lot of places where it has a `panic()`. Cleaning up all of the panics will be a work item in itself.